### ISSUE TITLE
Add Support for Displaying Matrices as Images

### ISSUE THREAD

**Issue by RoyiAvital** [j-fu/PlutoVista.jl#9]
Any chance to add an option to display a matrix or `m x n x c` array as image?
With the option to set the display size?

A native HTML mode would be great as well.

**Comment by j-fu** [comment]
Hi,isn't this already covered by Images.jl ? See https://github.com/fonsp/Pluto.jl/issues/60

With Images.jl and Colors.jl just can do `rand(RGB,200,200)` and the image is displayed in the notebook. With `reinterpret()` you can transform the matrix into this format (I think for this it needs to be `c x m x n`).

Also what do you mean by "native HTML mode" ?

**Comment by RoyiAvital** [comment]
I'd like support for images in the representation: `m x n x c` matrix and not only as the type defined in `Images.jl`.
In most other scientific frameworks (MATLAB, Python) images are just matrices in planar layout.

So supporting this would be great.

By native HTML I meant that the user could chose option to use the `img` tag in HTML (So behind the scene a binary stream will be created). Also, just an option.

The main motivation for this is able to display images in 1:1 (Pixel for pixel) form.

**Comment by j-fu** [comment]
I guess the image layout depends on row vs column major storage. I think that c x  m x n buffer in Julia has the very same memory layout as the n x m  x c buffer in numpy due to the fact that Julia is column major and numpy is row major. 

Native html using the img tag is not planned. And I bet one would have convert anyway to c x m x n to make this work, see above. And I guess that this is what images.jl does anyway.

The whole approach here is meant for smoot functions, and I cannot guarantee pixel 1:1.

**Comment by j-fu** [comment]
Hm ImageShow.jl uses the html <img> tag:

https://github.com/JuliaImages/ImageShow.jl/blob/245e0e4fa66001bd4d7b6f51b65b7d46844625cb/src/showmime.jl#L144

But as far as I understand now (my active html knowledge has quite some gaps, I read up on this) this must go throug base64 - which is a big performanc hurt.

So I see  the possibility to  draw the image directly onto canvas, passing binary data via `PlutoRunner.publish_to_js` to javascript. I indeed think that this would be faster than the stuff in Images.jl.

Without any transposition, the core API however would have to be  'c x m x n' because that is the way the data is  used in canvas - data for each pixel is stored contiguously:

https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/putImageData

Everything else would have to be copied to fit that layout. 

So yes in that respect this could fit here. But @fonsp has also some related stuff going on in PlutoUI.jl, this should be coordinated.

**Comment by j-fu** [comment]
When hovering is desired, it makes sense to work via plotly.js:

https://plotly.com/javascript/reference/image/#image

In fact, this appears to make much sense, as the infrastructure for the 
plotly backend is already here. I think I will go for that.  Have some homework on my desk to be done before starting though ...
